File: test-suite-docs-java/src/test/java/io/micronaut/mcp/docs/resources/PgnFile.java
package io.micronaut.mcp.docs.resources;

import io.micronaut.context.annotation.EachProperty;
import io.micronaut.context.annotation.Parameter;
import io.micronaut.context.annotation.Requires;
import io.micronaut.core.naming.Named;

/**
 * Configuration of a PGN file: {@code pgn.<name>.path} and {@code pgn.<name>.round}.
 */
@Requires(property = "spec.name", value = "ResourcesFactoryTest")
@EachProperty("pgn")
class PgnFile implements Named {
    private final String name;
    private String path;
    private Integer round;

    PgnFile(@Parameter String name) {
        this.name = name;
    }

    public Integer getRound() {
        return round;
    }

    public void setRound(Integer round) {
        this.round = round;
    }

    @Override
    public String getName() {
        return name;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }
}

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/resources/PgnFile.kt
package io.micronaut.mcp.docs.resources

import io.micronaut.context.annotation.EachProperty
import io.micronaut.context.annotation.Parameter
import io.micronaut.context.annotation.Requires
import io.micronaut.core.naming.Named

/**
 * Configuration of a PGN file: `pgn.<name>.path` and `pgn.<name>.round`.
 */
@Requires(property = "spec.name", value = "ResourcesFactoryTest")
@EachProperty("pgn")
class PgnFile(@param:Parameter private val name: String) : Named {
    var path: String? = null
    var round: Int? = null

    override fun getName(): String = name
}
